### PR TITLE
chore: update default case handling in facebook destinations

### PR DESCRIPTION
**Pull request by sanpj2292**
## What are the changes introduced in this PR?

For codes mentioned here, in default case, we will set message as `error_user_msg` instead of `message`
https://developers.facebook.com/docs/graph-api/guides/error-handling/

## What is the related Linear task?

Resolves INT-2969

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3908/test-report.html

**Review by sandeepdsvs — APPROVED** [review]

**Review by shrouti1507 — APPROVED** [review]

File: src/v0/util/facebookUtils/networkHandler.js [hunk 1]
```diff
@@ -107,66 +110,64 @@ const errorDetailsMap = {
...
-      .setMessageField('message')
+      .setMessageField('error_user_msg')
       .build(),
     default: new ErrorDetailsExtractorBuilder()
       .setStatus(400)
       .setStat({
         [TAG_NAMES.ERROR_TYPE]: ERROR_TYPES.AUTH,
       })
-      .setMessage('Invalid OAuth 2.0 access token')
+      .setMessageField('error_user_msg')
```
Comment (krishna2020) [inline src/v0/util/facebookUtils/networkHandler.js:128]: make sure `error_user_msg` indeed has more information that the manual error messages configured before

**Review by krishna2020 — APPROVED** [review]

File: src/v0/util/facebookUtils/networkHandler.js (hunk 1 above)
Comment (krishna2020) [reply src/v0/util/facebookUtils/networkHandler.js:128]: I don't see much changes in the network mocks with `error_user_msg` scenario

Comment (sanpj2292) [reply src/v0/util/facebookUtils/networkHandler.js:128]: > make sure error_user_msg indeed has more information that the manual error messages configured before

Yes. We have an example for the `Invalid Parameter` error, where in `error_user_msg` contains an error message to be shown for the user

> I don't see much changes in the network mocks with error_user_msg scenarios

Copied the message into `error_user_msg` field as didn't know what would the response be actually.
https://github.com/rudderlabs/rudder-transformer/pull/3908/files#diff-371cb884d55a278880bed7da54e3c846e7d83e4d0c07ad5d14d5fa6e25e5afb4R62

**Pull request merged**